src/index.ts

```diff
@@ -1,46 +1,122 @@
-import SSO from "./automations/SSO";
 import { JobInfo } from "./common/types";
 import regions from "./common/regions";
 import Job from "./automations/Job";
-import Puppeteer from "puppeteer";
-import { green } from "colors";
+import SSO from "./automations/SSO";
 import { Command, Argument, Option } from "commander";
+import { red } from "colors";
+import ora from "ora";
+// @ts-ignore This can be deleted after https://github.com/enquirer/enquirer/issues/135 is fixed.
+import { Select } from "enquirer";
 
-async function addPosts(jobID: number, regions: string[], cloneFrom: number) {
-    const sso = new SSO();
-    const loginCookies = await sso.login();
-    console.log(green("✓"), "Authentication complete");
+async function getJobFromCLI(job: Job, message: string, spinner: ora.Ora) {
+    // display UI
+    const jobs = await job.getJobs();
 
-    const browser = await Puppeteer.launch({ args: ["--no-sandbox"] });
-    const page = await browser.newPage();
-    await sso.setCookies(page, loginCookies);
+    spinner.succeed();
+    const prompt = new Select({
+        name: "Job",
+        message,
+        choices: [...jobs.keys()],
+    });
+
+    const jobName = await prompt.run();
+    return jobs.get(jobName);
+}
 
+async function addPosts(
+    isInteractive: boolean,
+    jobID: number,
+    regions: string[],
+    cloneFrom: number
+) {
+    const sso = new SSO();
+    const { browser, page } = await sso.authenticate();
     const job = new Job(page);
-    const jobData: JobInfo = await job.getJobData(jobID);
+    const spinner = ora("Fetching your jobs.").start();
+
+    try {
+        if (isInteractive) {
+            const id = await getJobFromCLI(
+                job,
+                "What job would you like to create job posts for?",
+                spinner
+            );
+            console.log(`Job ID: ${id}`);
+            // TODO add posts
+        } else {
+            if (!jobID) throw Error(`Job ID argument is missing.`);
+            if (!regions) throw Error(`Region parameter is missing.`);
+            const jobData: JobInfo = await job.getJobData(jobID);
 
-    // Process updates for each 'Canonical' job unless a "clone-from" argument is passed
-    await job.clonePost(jobData.posts, regions, cloneFrom);
+            // Process updates for each 'Canonical' job unless a "clone-from" argument is passed
+            await job.clonePost(jobData.posts, regions, cloneFrom);
```

Comment:
same here, update the spinner status all the time to let the user know what's happening

Reply:
:+1: